cmd/abapEnvironmentPullGitRepo.go

```diff
@@ -165,7 +166,15 @@ func getAbapCommunicationArrangementInfo(config abapEnvironmentPullGitRepoOption
 
 	if config.Host != "" {
 		// Host, User and Password are directly provided
-		connectionDetails.URL = "https://" + config.Host + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull"
+		matchedkey, error := regexp.MatchString(`^[hH][tT][tT][pP][sS]:\/\/.*`, config.Host)
+		if error != nil {
+			return connectionDetails, errors.New("Error occured while parsing the host parameter. Please check if this parameter has been seet correctly")
+		}
+		if matchedkey {
+			connectionDetails.URL = config.Host + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull"
+		} else {
+			connectionDetails.URL = "https://" + config.Host + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull"
```

Comment:
`config.Host` with a value of `http://example.org` (mind the missing "s") would result in `https://http://example.org` which is clearly not intended. On the other hand it is likely that this config-mistake will happen, so we should IMO prepared for that case and we should be able to handle that properly.

Reply:
The host of the ABAP Environment instance is quite difficult to type in (the format is like this: '1234-abcd-5678-efgh-ijk.abap.eu10.hana.ondemand.com'). Due to this, IMO the host will only be copied to the configuration.

For this reason, a likely error is to include "https://" with the host (one reason for this change), but both "http://" and "HTTPS://" is unlikely. 

On the other hand, I'm not sure how the URLs of onPrem Systems are handled (relevant for ATC) - maybe this invalidates my argument.

Reply:
Valid point. I discussed this with @DanielMieg earlier and we would view this as personal config-mistakes.
`config.Host` with value of `htps://example.org` (mind missing "t") would also result in misconfiguration like `https://htps://example.org`.

The current way is implicit, so the user will see failing config-mistake in logs when requesting the wrong URL. What you suggest is to explicitly add more cases to be checked.

Wouldnt cases like `htps`, `http`, `hps` and so on also have to be checked then? If requested this could be implemented...

Reply:
ooooook.